src/mscorlib/shared/System/IO/Win32Marshal.cs

```diff
@@ -2,106 +2,98 @@
             switch (errorCode)
             {
                 case Interop.Errors.ERROR_FILE_NOT_FOUND:
-                    if (path.Length == 0)
-                        return new FileNotFoundException(SR.IO_FileNotFound);
-                    else
-                        return new FileNotFoundException(SR.Format(SR.IO_FileNotFound_FileName, path), path);
-
+                    return string.IsNullOrEmpty(path)
+                        ? new FileNotFoundException(SR.IO_FileNotFound)
+                        : new FileNotFoundException(SR.Format(SR.IO_FileNotFound_FileName, path), path);
                 case Interop.Errors.ERROR_PATH_NOT_FOUND:
-                    if (path.Length == 0)
-                        return new DirectoryNotFoundException(SR.IO_PathNotFound_NoPathName);
-                    else
-                        return new DirectoryNotFoundException(SR.Format(SR.IO_PathNotFound_Path, path));
-
+                    return string.IsNullOrEmpty(path)
+                        ? new DirectoryNotFoundException(SR.IO_PathNotFound_NoPathName)
+                        : new DirectoryNotFoundException(SR.Format(SR.IO_PathNotFound_Path, path));
                 case Interop.Errors.ERROR_ACCESS_DENIED:
-                    if (path.Length == 0)
-                        return new UnauthorizedAccessException(SR.UnauthorizedAccess_IODenied_NoPathName);
-                    else
-                        return new UnauthorizedAccessException(SR.Format(SR.UnauthorizedAccess_IODenied_Path, path));
-
+                    return string.IsNullOrEmpty(path)
+                        ? new UnauthorizedAccessException(SR.UnauthorizedAccess_IODenied_NoPathName)
+                        : new UnauthorizedAccessException(SR.Format(SR.UnauthorizedAccess_IODenied_Path, path));
                 case Interop.Errors.ERROR_ALREADY_EXISTS:
-                    if (path.Length == 0)
+                    if (string.IsNullOrEmpty(path))
                         goto default;
-
                     return new IOException(SR.Format(SR.IO_AlreadyExists_Name, path), MakeHRFromErrorCode(errorCode));
-
                 case Interop.Errors.ERROR_FILENAME_EXCED_RANGE:
-                    if (path.Length == 0)
-                        return new PathTooLongException(SR.IO_PathTooLong);
-                    else
-                        return new PathTooLongException(SR.Format(SR.IO_PathTooLong_Path, path));
-
-                case Interop.Errors.ERROR_INVALID_DRIVE:
-                    throw new DriveNotFoundException(SR.Format(SR.IO_DriveNotFound_Drive, path));
-
-                case Interop.Errors.ERROR_INVALID_PARAMETER:
-                    return new IOException(Interop.Kernel32.GetMessage(errorCode), MakeHRFromErrorCode(errorCode));
-
+                    return string.IsNullOrEmpty(path)
+                        ? new PathTooLongException(SR.IO_PathTooLong)
+                        : new PathTooLongException(SR.Format(SR.IO_PathTooLong_Path, path));
```

Comment:
>path [](start = 85, length = 4)

Since it is path too long should we put an upper bound limit to the length on the exception?

Reply:
I don't think it is worth second guessing this one.

Reply:
hum actually the same concern applies elsewhere, that said, it is (in this regard) as it was before anyways...

---
In reply to: [175206161](https://github.com/dotnet/coreclr/pull/16998#discussion_r175206161) [](ancestors = 175206161)